isis/tests/BundleResultsTests.cpp

```diff
@@ -0,0 +1,547 @@
+  EXPECT_EQ(1000, results.numberUnknownParameters());
+  EXPECT_EQ(numImages, results.rmsImageSampleResiduals().size());
+  EXPECT_EQ(numImages, results.rmsImageLineResiduals().size());
+  EXPECT_EQ(numImages, results.rmsImageResiduals().size());
+  EXPECT_EQ(numImages, results.rmsLidarImageSampleResiduals().size());
+  EXPECT_EQ(numImages, results.rmsLidarImageLineResiduals().size());
+  EXPECT_EQ(numImages, results.rmsLidarImageResiduals().size());
+  EXPECT_EQ(0.5, results.minSigmaCoord1Distance().meters());
+  EXPECT_EQ(50, results.maxSigmaCoord1Distance().meters());
+  EXPECT_EQ(0.3, results.minSigmaCoord2Distance().meters());
+  EXPECT_EQ(30, results.maxSigmaCoord2Distance().meters());
+  EXPECT_EQ(0.1, results.minSigmaCoord3Distance().meters());
+  EXPECT_EQ(10, results.maxSigmaCoord3Distance().meters());
+  EXPECT_EQ("minCoord1Point", results.minSigmaCoord1PointId().toStdString());
+  EXPECT_EQ("maxCoord1Point", results.maxSigmaCoord1PointId().toStdString());
+  EXPECT_EQ("minCoord2Point", results.minSigmaCoord2PointId().toStdString());
+  EXPECT_EQ("maxCoord2Point", results.maxSigmaCoord2PointId().toStdString());
+  EXPECT_EQ("minCoord3Point", results.minSigmaCoord3PointId().toStdString());
+  EXPECT_EQ("maxCoord3Point", results.maxSigmaCoord3PointId().toStdString());
+  EXPECT_EQ(3.0, results.rmsRx());
+  EXPECT_EQ(4.0, results.rmsRy());
+  EXPECT_EQ(5.0, results.rmsRxy());
+  EXPECT_EQ(5.0, results.rejectionLimit());
+  EXPECT_EQ(0.5, results.sigma0());
+  EXPECT_EQ(180, results.elapsedTime());
+  EXPECT_EQ(75, results.elapsedTimeErrorProp());
+  EXPECT_EQ(5, results.iterations());
+  EXPECT_TRUE(results.converged());
+}
+
+
+TEST(BundleResults, Sigma0Computation) {
+  BundleResults results;
+  try {
+    results.computeSigma0(56.0, BundleSettings::Sigma0);
+    FAIL() << "Expected an exception";
+  }
+  catch (IException &e) {
+    EXPECT_THAT(
+          e.toString().toStdString(),
+          ::testing::HasSubstr("Computed degrees of freedom ["));
+  }
+  catch (...) {
+    FAIL() << "Expected an IException";
+  }
+  results.setNumberObservations(14);
+  results.computeDegreesOfFreedom();
+  results.computeSigma0(56.0, BundleSettings::Sigma0);
+  EXPECT_EQ(2.0, results.sigma0());
+
+  results.setNumberObservations(0);
+  results.computeDegreesOfFreedom();
+  results.computeSigma0(9.0, BundleSettings::ParameterCorrections);
+  EXPECT_EQ(3.0, results.sigma0());
+}
+
+
+TEST_F(BundleResultsPopulated, Save) {
+  QString saveString;
+  QXmlStreamWriter saveWriter(&saveString);
+  results.save(saveWriter, NULL);
+
+  EXPECT_FALSE(saveString.isEmpty());
+}
+
+
+TEST(BundleResults, NoOutputNet) {
+  BundleResults results;
+  try {
+    results.outputControlNet();
+    FAIL() << "Expected and exception";
+  }
+  catch (IException &e) {
+    EXPECT_THAT(
+          e.toString().toStdString(),
+          ::testing::HasSubstr("Output Control Network has not been set."));
+  }
+  catch (...) {
+    FAIL() << "Expected an IException";
+  }
+}
```

Comment:
Looking at the old truth data / unitTest.cpp did the mutator method and round trip serialization functionality get tested?

Reply:
The mutators are tested as a part of the fixture. The accessors test is testing both the mutators and accessors as you can't really test them independently without directly testing the internals of the class which is both not allowed and not a good idea in this type of test.